src/components/editor/EditorArticleController.jsx

```diff
@@ -0,0 +1,127 @@
+    this.handleMainFormChange = this.handleMainFormChange.bind(this);
+    this.authorListChanged = this.authorListChanged.bind(this);
+    this.safeSetState({
+      changed: "none",
+      saving: false
+    });
+  }
+
+  static getFalcorPath(params) {
+    return [['articlesBySlug', params.slug, ['title', 'issue', 'category', 'description']], ['articlesBySlug', params.slug, 'authors', {to: 2}, ['slug', 'name']]];
+  }
+
+  handleSaveChanges(event) {
+    event.preventDefault();
+  }
+
+  mainFormChanged(event) {
+    if (event.target.name === "addAuthor") {
+      return;
+    }
+
+    let changedFlag = this.handleMainFormChange();
+
+    let newChangedState = changedFlag ? "changed" : "none";
+    if (newChangedState !== this.state.changed) {
+      this.safeSetState({
+        changed: newChangedState
+      });
+    }
+  }
+
+  authorListChanged(authorsAdded, authorsDeleted) {
+    let changedFlag = Object.keys(authorsAdded).length > 0 || 
+      _.some(authorsDeleted, (val) => {
+        return val;
+    });
+    if (!changedFlag) {
+      changedFlag = changedFlag || this.handleMainFormChange();
+    }
+
+    let newChangedState = changedFlag ? "changed" : "none";
+    if (newChangedState !== this.state.changed) {
+      this.safeSetState({
+        changed: newChangedState
+      });
+    }
+  }
+
+  handleMainFormChange() {
+    let formNode = this.formNode;
+    let fields = ["category", "issue", "description"];
+    let falcorData = this.state.data.articlesBySlug[this.props.params.slug];
+
+    return fields.some((field) => {
+      let formValue = formNode[field].value;
+      if (field === "issue") formValue = parseInt(formValue);
+      let falcorValue = falcorData[field];
+
+      if (formValue !== falcorValue) {
+        return true;
+      }
+      return false;
+    });
+  }
+
+  bindFormNodeRef(ref) {
+    this.formNode = ref;
+  }
+
+  render() {
+    if (this.state.ready) {
+      if (!this.state.data) {
+        return <div><p>No articles match this slug</p></div>;
+      }
+
+      let slug = this.props.params.slug;
+      let data = this.state.data.articlesBySlug[slug];
+
+// The input type="number" behaves badly in Chrome, TODO: change to input type="text" and do validation yourself
+      return (
+        <div>
+          <h3>{data.title}</h3>
+          <p>Change the information for the article and press Save Changes to confirm the changes.</p>
+          <br/>
+          <form onChange={this.mainFormChanged} onSubmit={this.handleSaveChanges} ref={this.bindFormNodeRef.bind(this)}>
+            Change the Category:<br/>
+            <select defaultValue={data.category} name="category">
+              {categories.map((category) => {
+                return <option value={category} key={category}>{category}</option>;
+              })}
+            </select>
+            <div style={{height: "5px"}} />
+            Change Issue Number:<br/>
+            <input type="number" defaultValue={data.issue} name="issue" />
```

Comment:
okay. This is added to issues as well. Have to write a falcor path for it to be able to get the latest data